src/status/Status.ts

```diff
@@ -195,6 +196,16 @@ class Status {
     }
   }
 
+  @ready(new statusErrors.ErrorStatusNotRunning())
+  public async updateNodeId(newNodeId: NodeId): Promise<void> {
```

Comment:
I don't think we should have a separate method just to do this. It's better to expect that the our event handler just runs `this.writeStatus` directly. And yes, they will need to call `readStatus` beforehand to do so. So this logic should be put into your event handler.

Patching status is complicated by the fact that the SoT is on disk, and we don't have a type-safe way of representing it yet. So we can leave any development of `patchStatus` later.

Reply:
The `writeStatus` is currently `protected`, this allows for it work during `start` and `stop`.

So `writeStatus` is not able to be called outside.

Given that patching status only applies when the status is `LIVE`, I think this may mean we can have a function that only patches LIVE statuses.

Also we cannot do 1 read and 1 write. It has to be done within the same lock. So I'll have a new function that's called `updateStatusLive` that can only be used when status is `LIVE`.

Reply:
I've created an `updateStatusLive` that embeds the logic of `readStatus` and `writeStatus` together and ensures that the whole work is done within a single lock.

Reply:
Done and squashed into the second commit.